src/devices/machine/mb8421.h

```diff
@@ -135,22 +135,29 @@ class dual_port_mailbox_ram_base : public device_t
 
 	Type right_r(offs_t offset)
 	{
-		offset &= m_ram_mask;
+		offset &= RAM_MASK;
 		update_intr(read_or_write::READ, true, offset);
 		return m_ram[offset];
 	}
 
 	DECLARE_READ_LINE_MEMBER(busy_r) { return 0; } // _BUSY pin - not emulated
 
 protected:
-	dual_port_mailbox_ram_base(const machine_config &mconfig, device_type type, const char *tag, device_t *owner, u32 clock, size_t addr_bits, u8 data_bits);
+	dual_port_mailbox_ram_base(const machine_config &mconfig, device_type type, const char *tag, device_t *owner, u32 clock);
 
 	// device-level overrides
 	virtual void device_resolve_objects() override;
 	virtual void device_start() override;
 	virtual void device_reset() override;
 
+private:
 	// internal helpers
+	static constexpr Type DATA_MASK = make_bitmask<Type>(DataBits); // for DPRAMs with 9/18/36bit data buses
+	static constexpr size_t RAM_SIZE = make_bitmask<offs_t>(AddrBits); // max RAM word size
+	static constexpr size_t RAM_MASK = RAM_SIZE - 1;
+	static constexpr size_t INT_ADDR_LEFT = RAM_MASK - 1; // max RAM word size - 2
+	static constexpr size_t INT_ADDR_RIGHT = RAM_MASK; // max RAM word size - 1
```

Comment:
`RAM_MASK`, `INT_ADDR_LEFT` and `INT_ADDR_RIGHT` should be `offs_t` rather than `size_t`